tests/decomposition/test_decomposition_graph.py

```diff
@@ -91,6 +95,46 @@ def test_graph_construction(self, _):
         # 6 edges from ops to decompositions and 2 from decompositions to ops
         assert len(graph2._graph.edges()) == 8
 
+    def test_graph_construction_non_applicable_rules(self, _):
+        """Tests rules that raise DecompositionNotApplicable are skipped."""
+
+        class CustomOp(qml.operation.Operation):  # pylint: disable=too-few-public-methods
+            """A custom op"""
+
+            resource_keys = {"num_wires"}
+
+            @property
+            def resource_params(self):
+                return {"num_wires": len(self.wires)}
+
+        def _some_resource(num_wires):
+            if num_wires > 1:
+                raise DecompositionNotApplicable
+            return {qml.RZ: 1, qml.CNOT: 1}
```

Comment:
Could we include an example of using this in the Compiling circuits page? Where else could we document this?

Reply:
Maybe in the `Usage Details` of `register_resources`?